x-pack/plugins/alerting_builtins/server/alert_types/geo_threshold/es_query_builder.ts

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { ILegacyScopedClusterClient } from 'kibana/server';
+
+export const OTHER_CATEGORY = 'other';
+
+async function getShapesFilters(
+  boundaryIndexTitle: string,
+  boundaryType: string,
+  boundaryGeoField: string,
+  geoField: string,
+  callCluster: ILegacyScopedClusterClient['callAsCurrentUser'],
+  log: Logger
+) {
+  const filters = {};
+  switch (boundaryType) {
+    case 'entireIndex':
+      // Get all shapes in index
+      const boundaryData = await callCluster('search', {
+        index: boundaryIndexTitle,
+        body: {},
+      });
+      boundaryData.hits.hits.forEach(({ _index, _id }) => {
+        filters[_id] = {
+          geo_shape: {
+            [geoField]: {
+              indexed_shape: {
+                index: _index,
+                id: _id,
+                path: boundaryGeoField,
+              },
+            },
+          },
+        };
+      });
+      break;
+    default:
+      log.info(`Unsupported type: ${boundaryType}`);
```

Comment:
We've been trying to be in the habit of logging the alert type string and alert id string in messages like this, otherwise it can be hard to diagnose problems.

Reply:
Updated per your recommendations!